File: mario.py
try:
    height = int(input("Height: "))
except ValueError:
    height = input("Height: ")
    while height.isdigit() == False:
        height = input("Height: ")
while int(height) > 23:
    height = int(input("Height: "))
while int(height) < 1:
    height = int(input("Height: "))
rows = []
for i in range(height):
    rows.append([])
j = height - 1
k = 0
while j >= 0:
    blocksLeft = (height - j) + 1
    airLeft = j
    while airLeft > 0:
        rows[k].append(' ')
        airLeft -= 1
    while blocksLeft > 0:
        rows[k].append('#')
        blocksLeft -= 1
    j -= 1
    k += 1
for l in range(len(rows)):
    row = ''
    print(row.join(rows[l]))

File: mario.c
int height;
printf("Height: ")
scanf("%d", &height)
while(height < 1)
{
  printf("Height: ")
  scanf("%d", &height)
}
while(height > 23)
{
  printf("Height: ")
  scanf("%d", &height)
}
type rows[height];
int i = height - 1;
int j = 0;
while(i > 1)
{
  int blocksLeft = (height - i) + 1;
  int airLeft = i;
  while(airLeft > 1)
  {
    char air = " ";
    strncat(rows[i], &air, 1)
    airLeft --;
  }
  while(blocksLeft > 1)
  {
    char block = "#";
    strncat(rows[i], &block, 1)
    blocksLeft --;
  }
  i --;
}
for(j = 1; j <= height; i++)
{
  printf(rows[j], "\n")
}
